rasa/nlu/classifiers/keyword_intent_classifier.py

```diff
@@ -1,32 +1,99 @@
-from typing import Any, Optional, Text
+import os
+import logging
+import re
+import typing
+from typing import Any, Dict, Optional, Text
 
+from rasa.nlu import utils
 from rasa.nlu.components import Component
 from rasa.nlu.training_data import Message
 
+logger = logging.getLogger(__name__)
+
+if typing.TYPE_CHECKING:
+    from rasa.nlu.config import RasaNLUModelConfig
+    from rasa.nlu.training_data import TrainingData
+    from rasa.nlu.model import Metadata
+    from rasa.nlu.training_data import Message
+
 
 class KeywordIntentClassifier(Component):
+    """Intent classifier using simple keyword matching.
+
+
+    The classifier takes a list of keywords and associated intents as an input.
+    A input sentence is checked for the keywords and the intent is returned.
+
+    """
 
     provides = ["intent"]
 
-    his = ["hello", "hi", "hey"]
+    def __init__(self, intent_keyword_map: Optional[Dict] = None):
+
+        if intent_keyword_map is None:
+            self.intent_keyword_map = {}
+        else:
+            self.intent_keyword_map = intent_keyword_map
 
-    byes = ["bye", "goodbye"]
+    def train(
+        self,
+        training_data: "TrainingData",
+        cfg: Optional["RasaNLUModelConfig"] = None,
+        **kwargs: Any
+    ) -> None:
+        self.intent_keyword_map = {}
+        for intent in training_data.intents:
+            self.intent_keyword_map[intent] = [
+                ex.text
+                for ex in training_data.training_examples
+                if ex.get("intent") == intent
+            ]
 
     def process(self, message: Message, **kwargs: Any) -> None:
+        intent_name = self.map_keyword_to_intent(message.text)
+        if intent_name is not None:
+            intent = {"name": intent_name, "confidence": 1.0}
+            message.set("intent", intent, add_to_output=True)
 
-        intent = {"name": self.parse(message.text), "confidence": 1.0}
-        message.set("intent", intent, add_to_output=True)
+    def map_keyword_to_intent(self, text: Text) -> Optional[Text]:
 
-    def parse(self, text: Text) -> Optional[Text]:
+        for intent, keywords in self.intent_keyword_map.items():
+            for string in keywords:
+                if re.search(r"\b" + string + r"\b", text):
```

Comment:
does it works if string contains spaces?

Reply:
Yes, it returns the intent iff the entered text is the same as one of the training examples. 
Actually the regex was needed when we were still searching for keywords, now that it's just string matching i think `==` will suffice

Reply:
hm, but then `yes, blabla` will not be classified as yes, is it intended?

Reply:
probably better to just use `in`